Libraries/Geolocation/RCTLocationObserver.m

```diff
@@ -68,6 +68,18 @@ + (RCTLocationOptions)RCTLocationOptions:(id)json
 
 @end
 
+@implementation RCTConvert (CoreLocation)
+
+RCT_ENUM_CONVERTER(CLAuthorizationStatus, (@{
```

Comment:
Should we unify these names with Android counterpart... somehow?

Reply:
I actually tried that in the app where I extracted this methods from, and that didn't went well.. That might of course have just be me. 

But for one the status of the authorization is binary on Android, or at least with the currently available API:

```js
PermissionsAndroid.check(permission).then((granted) => {
  if (granted) {
    // ...
  }
});
```

Where on iOS there a lot more values.

Also now that I think of it, I should probably also add `getAuthorizationStatus` to this PR.

Reply:
Yeah, I see. So, theoretically, we can use iOS values on Android (something like `always` and `denied`), right?

Reply:
Yes and no, because the API for android will return `false` when the user was never prompted a request in the first place, whereas iOS then returns `notDetermined`.

Reply:
However, on requesting permissions more details are revealed; 

```js
const permission = PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION;
PermissionsAndroid.request(permission).then((result) => {
  if (result === PermissionsAndroid.RESULTS.NEVER_ASK_AGAIN) {
    //
  }
  if (result === PermissionsAndroid.RESULTS.DENIED) {
    //
  }
  if (result === PermissionsAndroid.RESULTS.GRANTED) {
    //
  }
});

Reply:
Maybe we can add something [like this](https://github.com/facebook/react-native/blob/75123c614bb5a61c383cdc247230b3a76c76e14d/ReactAndroid/src/main/java/com/facebook/react/modules/permissions/PermissionsModule.java#L117-L124) to the `check` method, although this would be a breaking change on the API..

Reply:
Or maybe we can implement the `getAuthorizationStatus` also for Android, and leave the `PermissionsAndroid.check` as-is?